### PR TITLE
Feature/avro serialization

### PR DESCRIPTION
**Pull request by frankwis**

### CODE REVIEW

File: src/main/java/io/datanerds/avropatch/Serializer.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+public class Serializer {
+
+    private static final Schema schema = ReflectData.get().getSchema(Patch.class);
+    private static final DatumWriter<Patch> writer = new ReflectDatumWriter<>(schema);
+
+    public static byte[] toBytes(Patch patch) throws IOException {
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        Encoder binaryEncoder = EncoderFactory.get().directBinaryEncoder(outputStream, null);
+
+        writer.write(patch, binaryEncoder);
```
Comment (larsp) [inline src/main/java/io/datanerds/avropatch/Serializer.java]: I think you already told me that, but write() is thread-safe?

File: src/main/java/io/datanerds/avropatch/operation/Add.java [hunk 2]
```diff
@@ -3,13 +3,16 @@
...
-public class Add<T> implements Operation {
+public final class Add<T> implements Operation {
     @AvroIgnore
     public static final String op = "add";
     public final Path path;
     @AvroSchema(Value.SCHEMA)
     public final T value;
 
+    /**
+     * Private no arg constructor for Avro deserializer.
```
Comment (larsp) [inline src/main/java/io/datanerds/avropatch/operation/Add.java]: Why this javadoc?

File: src/main/java/io/datanerds/avropatch/operation/Copy.java [hunk 3]
```diff
@@ -1,21 +1,22 @@
...
-    public Copy(Path path, Path from) {
-        this.path = path;
+    /**
+     * Private no arg constructor for Avro deserializer.
+     */
+    private Copy() {
+        from = null;
+        path = null;
+    }
+    public Copy(Path from, Path path) {
```
Comment (larsp) [inline src/main/java/io/datanerds/avropatch/operation/Copy.java:26]: public constructor has no javadoc?

File: src/test/java/io/datanerds/avropatch/operation/matcher/OperationMatchers.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+import java.util.*;
+import java.util.function.Function;
+
+import static org.hamcrest.core.AllOf.allOf;
+
+public final class OperationMatchers {
+
+    private static Map<Class<? extends Operation>, Function<Operation, Matcher<? extends Operation>>> map = new HashMap<>();
+    static {
+        map.put(Add.class, expected -> Matchers.equalTo((Add)expected));
```
Comment (larsp) [inline src/test/java/io/datanerds/avropatch/operation/matcher/OperationMatchers.java]: formatting looks off

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+language: java
```
Comment (larsp) [inline .travis.yml:1]: works! 👍

**Review by larsp — CHANGES_REQUESTED** [review]
minor comments :shipit:

File: src/main/java/io/datanerds/avropatch/Serializer.java (hunk 1 above)
Comment (frankwis) [reply src/main/java/io/datanerds/avropatch/Serializer.java]: Yes, it is :-)

**Review by larsp — APPROVED** [review]

**Pull request merged**